Assets/Scripts/BuyUpgrade.cs

```diff
@@ -22,10 +24,11 @@ public void trybuyUpdate(int count)
         Message.Send(new Upgrade(1));
     }
 
-    void CanHasUpgrade(CoinsUpdate msg)
+    void CanHasUpgrade(CoinsUpdate msg, Price msg2)
     {
+        price = msg2.price;
         count = msg.count;
-        if (count < 10)
+        if (count < price)
```

Comment:
Code Control events only take one argument. A class that inherits from Message. 

You can't have two different events call the same event handler like this.
They can call different handlers that call the same function. 

Split these into 3 functions. 
OnCoinsUpdate(CoinsUpdate msg)
OnPriceUpdate(PriceUpdate msg)
CanHasUpgrade()